server/src/main/java/org/elasticsearch/search/aggregations/metrics/MinAggregator.java

```diff
@@ -210,6 +212,7 @@ static Number findLeafMinValue(LeafReader reader, String fieldName, Function<byt
             return converter.apply(pointValues.getMinPackedValue());
         }
         final Number[] result = new Number[1];
+        final AtomicInteger lookupCounter = new AtomicInteger(0);
```

Comment:
I could be wrong, but I think we can move the counter inside the `PointValues.IntersectVisitor()` anonymous class and avoid having to use an atomic.  I'll defer to @iverase though, in case intersect behaves differently than I'm thinking.

Reply:
Great, atomic replaced with `short` field in the visitor.